src/libraries/Microsoft.PowerFx.Connectors/OpenApiParser.cs

```diff
@@ -343,12 +359,21 @@ internal static List<ServiceFunction> Parse(string functionNamespace, OpenApiDoc
                 throw new ArgumentException(nameof(functionNamespace));
             }
 
-            ValidateSupportedOpenApiDocument(openApiDocument);
+            connectorSettings ??= new ConnectorSettings();
+            ValidateSupportedOpenApiDocument(openApiDocument, connectorSettings.IgnoreUnknownExtensions);
 
             List<ServiceFunction> functions = new List<ServiceFunction>();
             string basePath = openApiDocument.GetBasePath();
-            DPath theNamespace = DPath.Root.Append(new DName(functionNamespace));
-            connectorSettings ??= new ConnectorSettings();
+
+            // $$$ basePath is just "/", but we expect it to be 'server' from the swagger file. 
+            // eg, "https://api.math.tools"
+
+            DPath theNamespace = DPath.Root.Append(new DName(functionNamespace));            
+
+            if (httpClient != null && httpClient is HttpClient hc && hc.BaseAddress == null && openApiDocument.Servers.Any())
+            {
+                hc.BaseAddress = new Uri(openApiDocument.Servers.First().Url);
+            }
```

Comment:
@LucGenetier  - this seems dangerous. 
It's mutating the incoming httpClient, which could be shared.